### ISSUE TITLE
Proposal: swap [size] and [cond] in CMOV[size][cond]

### ISSUE THREAD

**Issue by dlespiau** [golang/go#20173]
This proposal is originally from @rsc in issue #14069 and is about `CMOVLEQ` being ambiguous. I Extracted it as I believe it deserves its own issue.

> Recounting this to @aclements I figured out what is going on with CMOVLEQ. The GNU form is CMOV[cond][size] but the form added for Go is CMOV[size][cond]. So GNU's cmovleq is cmov-le-q while Go's CMOVLEQ is CMOV-L-EQ. The Intel syntax has no size suffix and is CMOVEQ/CMOVLE, but it seems clear that Go should not be inserting a size suffix in the middle of the defined opcode name. While in general I feel that we need to put up with past mistakes in our assembly definitions to avoid breaking existing assembly programs, this one seems so egregious and error-prone that I think we have no choice but to change these names for Go 1.7 to match the Intel and GNU syntax.

The hard bit is to decide if this is really worth it and how to manage the breakage this would incur:
1. Projects that want to compile with go versions before and after that change would need to carry two versions of the assembler code
2. CMOVLEQ being ambiguous, if a source file only contains this CMOV form it's not possible to tell if it has been fixed or not

I have a small shell script that does the change across the whole `src/` directory and the result passes all.bash.
```
#!/bin/sh
        
# matches CMOV instructions for sed -r and git grep -E
CMOV='CMOV([BWLQ])(HI|CC|CS|HS|LS|QW|EQ|GT|GE|LT|LE|NE|OC|PC|PL|OS|PS|MI)'
files=`git grep -E $CMOV  | cut -d : -f 1 | sort -u`

sed -i -r "s/$CMOV/CMOV\2\1/g" $files
```
This changes CMOV for x86 and ARM.

**Idea**: It would be possible to include that rewrite in `go fix`.

**Optional** (and orthogonal actually): Define aliases with Intel conditions. Just like jumps today it may be handy for people looking at the Intel manual or porting code from other projects to have the same `cc` abbreviations.

**Comment by rsc** [comment]
At this point I don't think we can rename instructions like this - it will silently change the meaning of existing (one assumes debugged) code. If we need to do something at all, I would suggest adding CMOVL.EQ as an alias for current Go CMOVLEQ, and so on, and then in a later release removing the dot-free variants.

**Comment by rsc** [comment]
No one replied to previous comment (but two thumbs up) so I'm repurposing for that "clearer new spelling" instead of "change existing meaning" fix.

**Comment by dlespiau** [comment]
Yes, thank you for renaming the issue. Updated the proposal text accordingly as well.

**Comment by Quasilyte** [comment]
Should this be done for 1.11?

Suffixes are used for AVX-512 things, should we reconsider `CMOVL.EQ` or add additional suffixes for these? (see #22779)

**Comment by Symmtek** [comment]
> Unfortunately it seems too late for such a breaking change so the proposal evolved into introducing new aliases like CMOVL.EQ (CMOV%size.%cond) to lift the ambiguity and then, later on, remove the old versions.

Added an alias `CMOVL.EQ` for `CMOVLEQ`. Should probably remove old version in the future.
